src/pemfile.rs

```diff
@@ -1,36 +1,37 @@
 use std::io::{self, ErrorKind};
+use std::iter;
+
+use pki_types::{
+    CertificateDer, CertificateRevocationListDer, PrivatePkcs1KeyDer, PrivatePkcs8KeyDer,
+    PrivateSec1KeyDer,
+};
 
 /// The contents of a single recognised block in a PEM file.
 #[non_exhaustive]
 #[derive(Debug, PartialEq)]
 pub enum Item {
     /// A DER-encoded x509 certificate.
-    X509Certificate(Vec<u8>),
-
-    /// A DER-encoded plaintext RSA private key; as specified in PKCS#1/RFC3447
-    RSAKey(Vec<u8>),
-
-    /// A DER-encoded plaintext private key; as specified in PKCS#8/RFC5958
-    PKCS8Key(Vec<u8>),
-
-    /// A Sec1-encoded plaintext private key; as specified in RFC5915
-    ECKey(Vec<u8>),
-
-    /// A Certificate Revocation List; as specified in RFC5280
-    Crl(Vec<u8>),
-}
-
-impl Item {
-    fn from_start_line(start_line: &[u8], der: Vec<u8>) -> Option<Item> {
-        match start_line {
-            b"CERTIFICATE" => Some(Item::X509Certificate(der)),
-            b"RSA PRIVATE KEY" => Some(Item::RSAKey(der)),
-            b"PRIVATE KEY" => Some(Item::PKCS8Key(der)),
-            b"EC PRIVATE KEY" => Some(Item::ECKey(der)),
-            b"X509 CRL" => Some(Item::Crl(der)),
-            _ => None,
-        }
-    }
+    ///
+    /// Appears as "CERTIFICATE" in PEM files.
+    X509Certificate(CertificateDer<'static>),
+
+    /// A DER-encoded plaintext RSA private key; as specified in PKCS #1/RFC 3447
+    ///
+    /// Appears as "RSA PRIVATE KEY" in PEM files.
+    Pkcs1Key(PrivatePkcs1KeyDer<'static>),
+
+    /// A DER-encoded plaintext private key; as specified in PKCS #8/RFC 5958
+    ///
+    /// Appears as "PRIVATE KEY" in PEM files.
+    Pkcs8Key(PrivatePkcs8KeyDer<'static>),
+
+    /// A Sec1-encoded plaintext private key; as specified in RFC 5915
+    ///
+    /// Appears as "EC PRIVATE KEY" in PEM files.
+    Sec1Key(PrivateSec1KeyDer<'static>),
+
+    /// A Certificate Revocation List; as specified in RFC 5280
```

Comment:
Nit: how about adding a `// Appears as "X509 CRL" in PEM files.` here too? ([RFC 7468, Section 4](https://www.rfc-editor.org/rfc/rfc7468#section-4))

Reply:
@djc Did the update for this get lost in the shuffle? The conversation was marked resolved but the comment looks unchanged.

Reply:
Oops, looks like I forgot! Should be fixed now.

Reply:
Thanks!